okhttp-android-support/src/test/java/com/squareup/okhttp/internal/huc/ResponseCacheTest.java

```diff
@@ -1866,8 +1866,9 @@ private void assertGatewayTimeout(HttpURLConnection connection) throws IOExcepti
         HttpURLConnection httpURLConnection = (HttpURLConnection) connection;
         assertEquals(server.url("/").url(), uri.toURL());
         assertEquals(200, httpURLConnection.getResponseCode());
+        InputStream is = httpURLConnection.getInputStream();
         try {
-          httpURLConnection.getInputStream();
```

Comment:
That fails 100s of tests since it is now eagerly opened above. The alternative is a `Source` that lazily opens the underlying `InputStream` on first access and then forwards all calls.